# This patch is needed for ISPC build with LLVM 7.0 for windows.
# Without this patch, files compiled using ISPC will fail with a linking
# error on windows. This change is based on the commit from LLVM revision
# r339307.
Index: cmake/modules/GetHostTriple.cmake
===================================================================
--- cmake/modules/GetHostTriple.cmake	(revision 346555)
+++ cmake/modules/GetHostTriple.cmake	(working copy)
@@ -4,15 +4,15 @@
 function( get_host_triple var )
   if( MSVC )
     if( CMAKE_SIZEOF_VOID_P EQUAL 8 )
-      set( value "x86_64-pc-win32" )
+      set( value "x86_64-pc-windows-msvc" )
     else()
-      set( value "i686-pc-win32" )
+      set( value "i686-pc-windows-msvc" )
     endif()
   elseif( MINGW AND NOT MSYS )
     if( CMAKE_SIZEOF_VOID_P EQUAL 8 )
-      set( value "x86_64-w64-mingw32" )
+      set( value "x86_64-w64-windows-gnu" )
     else()
-      set( value "i686-pc-mingw32" )
+      set( value "i686-pc-windows-gnu" )
     endif()
   else( MSVC )
     set(config_guess ${LLVM_MAIN_SRC_DIR}/cmake/config.guess)
